ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -375,15 +375,44 @@ private void onAccountChangeMessage(String topic, String message) {
    */
   public void selectInactiveContainersAndMarkInZK(StatsSnapshot statsSnapshot) {
     Set<Container> inactiveContainerCandidateSet = selectInactiveContainerCandidates(statsSnapshot);
-    markContainerInactiveOnZk(inactiveContainerCandidateSet);
+    try {
+      markContainerInactiveOnZk(inactiveContainerCandidateSet);
+    } catch (InterruptedException e) {
+      logger.error("Mark inactive container in zookeeper is interrupted", e);
+    }
   }
 
   /**
    * Mark the given {@link Container}s status to INACTIVE in zookeeper.
    * @param inactiveContainerCandidateSet DELETE_IN_PROGRESS {@link Container} set which has been deleted successfully during compaction.
    */
-  private void markContainerInactiveOnZk(Set<Container> inactiveContainerCandidateSet) {
-    // TODO: mark the given containers status to INACTIVE in zookeeper.
+  void markContainerInactiveOnZk(Set<Container> inactiveContainerCandidateSet) throws InterruptedException {
+    if (inactiveContainerCandidateSet != null) {
+      boolean successful = false;
+      int retry = 0;
+      while (!successful && retry < config.retryCount) {
+        Map<Short, Account> accountToUpdateMap = new HashMap<>();
+        inactiveContainerCandidateSet.forEach(container -> {
+          // start by getting account, and then get container from account to make sure that we are editing the most
+          // recent snapshot
+          short accountId = container.getParentAccountId();
+          Account accountToEdit = accountToUpdateMap.computeIfAbsent(accountId, this::getAccountById);
+          Container containerToEdit = accountToEdit.getContainerById(container.getId());
+          Container editedContainer =
+                  new ContainerBuilder(containerToEdit).setStatus(Container.ContainerStatus.INACTIVE).build();
+          accountToUpdateMap.put(accountId,
+                  new AccountBuilder(accountToEdit).addOrUpdateContainer(editedContainer).build());
+        });
+        successful = updateAccounts(accountToUpdateMap.values());
+        if (!successful) {
+          retry++;
+          Thread.sleep(config.retryDelay);
+        }
+      }
+      if (!successful) {
+        logger.error("Container failed to be marked as INACTIVE in inactiveContainerCandidateSet : {}", inactiveContainerCandidateSet);
```

Comment:
```
logger.error("Failed to mark containers INACTIVE in set : {}  after {} retries", inactiveContainerCandidateSet, config.retryCount);
```

Reply:
Also I am considering if we need a metric here for alerting purpose.

Reply:
Updated and add the metric.